odoo/addons/base/models/ir_model.py

```diff
@@ -597,7 +597,12 @@ def _in_modules(self):
     @api.constrains('domain')
     def _check_domain(self):
         for field in self:
-            safe_eval(field.domain or '[]')
+            try:
+                safe_eval(field.domain or '[]')
+            except Exception:
+                raise ValidationError(
+                    _("Invalid Python code '%s' for the domain field.", field.domain)
+                ) from None
```

Comment:
Can you add the reason for the failure ? It's still a technical view, the exception message can be very helpful.

```suggestion
            except Exception as e:
                raise ValidationError(
                    _("Invalid Python code '%s' for the domain field: %s", field.domain, e)
                ) from None
```